lib/ForEach.js

```diff
@@ -20,12 +20,16 @@ function ForEach() {
  * Fallback for glTF 1.0
  * @private
  */
-ForEach.objectLegacy = function(objects, handler) {
+ForEach.objectLegacy = function (objects, handler) {
     if (defined(objects)) {
         for (var objectId in objects) {
             if (objects.hasOwnProperty(objectId)) {
                 var object = objects[objectId];
-                handler(object, objectId);
+                var value = handler(object, objectId);
+
+                if (defined(value)) {
+                    return value;
```

Comment:
Are there any area in gltf-pipeline that can take advantage of early returns now? https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/2.0/lib/writeResources.js#L141-L152 comes to mind. If anything else comes up we can just fix as we notice them.